src/lib/components/ui/input/date-input.svelte

```diff
@@ -1,8 +1,352 @@
+		)
+	}
+
+	function formatDate(date: Date) {
+		const year = date.getFullYear()
+		const month = String(date.getMonth() + 1).padStart(2, '0')
+		const day = String(date.getDate()).padStart(2, '0')
+		return `${year}-${month}-${day}`
+	}
+
+	function parseDate(dateString: string) {
+		const [year, month, day] = dateString.split('-').map(Number)
+		return new Date(year, month - 1, day)
+	}
+
+	function inputChange(event: Event) {
+		const input = event.target as HTMLInputElement
+		const newDate = parseDate(input.value)
+		selectedDate = newDate
+		selectedMonth = newDate.getMonth()
+		selectedYear = newDate.getFullYear()
+		showYearPicker = false
+	}
+
+	function close(e: MouseEvent) {
+		const target = e.target as unknown as Node
+		if (datePicker?.contains(target)) {
+			// Clicked on the date picker
+		} else {
+			showDatePicker = false
+			showYearPicker = false
+		}
+	}
+
+	$effect(() => {
+		window.addEventListener('click', close)
+		return () => {
+			window.removeEventListener('click', close)
+		}
+	})
+
+	$effect(() => {
+		selectedYear = selectedDate.getFullYear()
+		selectedMonth = selectedDate.getMonth()
+	})
+
+	$effect(() => {
+		if (showYearPicker) {
+			selectedYearElement?.scrollIntoView({ block: 'center', behavior: 'auto' })
+		}
+	})
 </script>
 
-<Input type="date" {...restProps} />
+{#snippet buttons()}
+	<Button
+		{dimension}
+		{disabled}
+		variant="secondary"
+		onclick={(e:MouseEvent) => {
+			e.stopPropagation()
+			showDatePicker = !showDatePicker
+			showYearPicker = false
+		}}
+	>
+		<Calendar {size} />
+	</Button>
+{/snippet}
+
+<div class="calendar-root {dimension}">
+	<Input
+		{dimension}
+		{disabled}
+		{...restProps}
+		value={formatDate(selectedDate)}
+		oninput={inputChange}
+		{buttons}
+		type="date"
+	/>
+	<div class="date-picker" class:showDatePicker bind:this={datePicker}>
+		<div class="header">
+			<div class="month">
+				<Button {dimension} variant="ghost" onclick={() => changeMonth(-1)}
+					><ChevronLeft {size} /></Button
+				>
+				<div class="current-month">
+					<Typography {variant} bold>{months[selectedMonth]} {selectedYear}</Typography>
+					<Button
+						{dimension}
+						variant="ghost"
+						onclick={(e:MouseEvent)=>{e.stopPropagation(); showYearPicker=!showYearPicker}}
+					>
+						{#if showYearPicker}
+							<CaretUp {size} />
+						{:else}
+							<CaretDown {size} />
+						{/if}
+					</Button>
+				</div>
+				<Button {dimension} variant="ghost" onclick={() => changeMonth(1)}
+					><ChevronRight {size} /></Button
+				>
+			</div>
+			{#if !showYearPicker}
+				<div class="days">
+					{#each days as day}
+						<Typography {variant}>{day}</Typography>
+					{/each}
+				</div>
+			{/if}
+		</div>
+
+		<Divider class="no-margin" />
+
+		{#if showYearPicker}
+			<div class="year-picker">
+				<!-- eslint-disable-next-line @typescript-eslint/no-unused-vars -->
+				{#each Array(131) as _, i}
+					{#if selectedYear === 1970 + i}
```

Comment:
Since the date is often used for birth dates, I would change this to 1900.

The ideal solution is that the start date is not hardcoded, but dynamically calculated when pressing the back button.